components/Article/TableOfContents/index.tsx

```diff
@@ -0,0 +1,43 @@
+import { FormattedMessage } from 'react-intl';
+import Link from 'next/link';
+import styles from './index.module.scss';
+import type { FC } from 'react';
+
+type PropsTableOfContents = {
+  depth: number;
+  value: string;
+  id: string;
+}[];
+
+const traverseTableOfContents = (tableOfContents: PropsTableOfContents) => {
+  return (
+    <ul>
+      {tableOfContents.map(item => {
+        return (
+          <li key={item.id}>
+            <Link href={`#${item.id}`}>{item.value}</Link>
+          </li>
+        );
+      })}
+    </ul>
+  );
+};
```

Comment:
```suggestion
const traverseTableOfContents = (tableOfContents: PropsTableOfContents) => (
  <ul>
    {tableOfContents.map(item => (
      <li key={item.id}>
        <Link href={`#${item.id}`}>{item.value}</Link>
      </li>
    ))}
  </ul>
);

```
You don't need `return` statements. 


Also, can you convert this to a Component instead of a Function. It's considered a bad pattern.

Reply:
Thanks for giving this advidce but on original component we have nested list.
I have to update the component

Reply:
Now take a look, `return` is needed

Reply:
Why is return needed?

Reply:
Look up the code now, you will have the answer 😁

Reply:
@AugustinMauroy As Harkunwar mentioned this function can be changed to React component

Reply:
I don't see a problem of using `return` here but if you really want to remove it you can change the following line
```diff
- {MappedTableOfContents.map(item => (
+ {parseTableOfContents(tableOfContents).map(item => (
```

Reply:
Good idea !